jetty-cdi/src/main/java/org/eclipse/jetty/cdi/CdiDecorator.java

```diff
@@ -0,0 +1,172 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.cdi;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.jetty.util.Decorator;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.webapp.WebAppContext;
+
+/**
+ * A Decorator invokes a CDI provider within
+ * a webapplication to decorate objects created by the contexts {@link org.eclipse.jetty.util.DecoratedObjectFactory}
+ * (typically Listeners, Filters and Servlets).
+ * The CDI provide is invoked using {@link MethodHandle}s to avoid any CDI instance or dependencies within the server scope.
+ * The code invoked is equivalent to:
+ * <pre>
+ * public &lt;T&gt; T decorate(T o)
+ * {
+ *   BeanManager manager = CDI.current().getBeanManager();
+ *   manager.createInjectionTarget(manager.createAnnotatedType((Class&lt;T&gt;)o.getClass()))
+ *     .inject(o,manager.createCreationalContext(null));
+ *   return o;
+ * }
+ * </pre>
+ */
+public class CdiDecorator implements Decorator
+{
+    private static final Logger LOG = Log.getLogger(CdiServletContainerInitializer.class);
+
+    private final WebAppContext _context;
+    private final Class<?> _cdiClass;
+    private final Class<?> _beanManagerClass;
+    private final Class<?> _annotatedTypeClass;
+    private final Class<?> _injectionTargetClass;
+    private final Class<?> _creationalContextClass;
+    private final Class<?> _contextualClass;
```

Comment:
All these `Class<?>` fields can be converted to local variables, no need to keep them as fields.